services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -15,38 +18,93 @@ object ApplicationDataMapper {
         applicationData: CableReportApplicationData
     ): AlluCableReportApplicationData =
         with(applicationData) {
-            val description = workDescription(cableReport = this)
+            val description =
+                workDescription(workDescription, rockExcavation.orThrow("rockExcavation"))
             AlluCableReportApplicationData(
+                postalAddress = postalAddress?.toAlluData(),
                 name = name,
                 customerWithContacts =
                     customerWithContacts
                         .orThrow(path("customerWithContacts"))
                         .toAlluData(path("customerWithContacts")),
-                geometry = getGeometry(applicationData = this),
+                representativeWithContacts =
+                    representativeWithContacts?.toAlluData(path("representativeWithContacts")),
+                invoicingCustomer = null,
+                geometry = getGeometry(this),
                 startTime = startTime.orThrow(path("startTime")),
                 endTime = endTime.orThrow(path("endTime")),
                 pendingOnClient = pendingOnClient,
                 identificationNumber = hankeTunnus,
+                customerReference = null,
+                area = null,
+                trafficArrangementImages = null,
                 clientApplicationKind = description, // intentional
                 workDescription = description,
+                propertyDeveloperWithContacts =
+                    propertyDeveloperWithContacts?.toAlluData(
+                        path("propertyDeveloperWithContacts")
+                    ),
                 contractorWithContacts =
                     contractorWithContacts
-                        .orThrow(path("contractorWithContacts"))
+                        .orThrow("contractorWithContacts")
                         .toAlluData(path("contractorWithContacts")),
-                postalAddress = postalAddress?.toAlluData(),
+                constructionWork = constructionWork,
+                maintenanceWork = maintenanceWork,
+                emergencyWork = emergencyWork,
+                propertyConnectivity = propertyConnectivity
+            )
+        }
+
+    fun toAlluData(
+        hankeTunnus: String,
+        applicationData: ExcavationNotificationApplicationData
+    ): AlluExcavationNotificationApplicationData =
+        with(applicationData) {
+            val description =
+                workDescription(
+                    workDescription,
+                    rockExcavation ?: false
+                ) // TODO as in cable report?
```

Comment:
Excavation notification doesn't need the rock excavation -information. It's just there in case the customer wants to create a cable report application with the same form.

Reply:
You're right. I changed the logic of how work description text is formed. This might change if the whole description logic differs from cable report.